src/Illuminate/View/Compilers/Concerns/CompilesErrors.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Illuminate\View\Compilers\Concerns;
+
+trait CompilesErrors
+{
+    /**
+     * Compile the error statements into valid PHP.
+     *
+     * @param  string  $expression
+     * @return string
+     */
+    protected function compileError($expression)
+    {
+        $expression = $this->stripParentheses($expression);
+
+        return '<?php if ($errors->has('.$expression.')) :
+if (isset($message)) { $messageCache = $message; }
+$message = $errors->first('.$expression.'); ?>';
+    }
+
+    /**
+     * Compile the enderror statements into valid PHP.
+     *
+     * @param  string  $expression
+     * @return string
+     */
+    protected function compileEnderror($expression)
+    {
+        return '<?php unset($message);
+if (isset($messageCache)) { $message = $messageCache; }
```

Comment:
Why would you need to nest? You can use dot format to render deeply nested error messages.

Reply:
Sorry, I said "believe" not to be confrontational, but to express I don't actually know for sure. I meant "believe" to be a loose word, not strong lol.

I'm pretty indifferent on supporting nesting because I can't really see myself using it. But I agree it seems like the behavior is unexpected if what you're saying is true.

Can you provide the solution?

Reply:
> Can you provide the solution?

Just utilize `array_pop`/`array_push` and make `$messageCache` into array instead. This will look something like this:

```php
    // For opening directive
    return '<?php if ($errors->has('.$expression.')) :
        if (! isset($message)) { $messageCache = array(); }
        array_push($messageCache, $message);
        $message = $errors->first('.$expression.'); ?>';
    
    ...
    
    // For closing one
    return '<?php unset($message);
        if (! empty($messageCache)) { $message = array_pop($messageCache); }
```

But make sure to cover it with separate test if you do decide to get rid of this problem and introduce nesting support without potential side-effects.